datadelivery/config.py

```diff
@@ -47,18 +48,38 @@ def read_config(self):
         with open(self.filename, 'r') as stream:
             return Config(yaml.safe_load(stream))
 
-    def write_new_config(self, token):
+    def write_config(self, config):
         with open(self.filename, 'w+') as stream:
-            yaml.safe_dump({
-                'token': token
-            }, stream)
+            yaml.safe_dump(config.to_dict(), stream)
 
 
 class Config(object):
     def __init__(self, data):
-        self.token = data['token']
-        self.url = data.get('url', DEFAULT_DATA_DELIVERY_URL)
-        self.endpoint_name = data.get('endpoint_name', DEFAULT_ENDPOINT_NAME)
+        self.token = data.get('token')
```

Comment:
Were these changed to lazily-loaded properties to avoid writing out defaults to the YAML file?

Reply:
Yes.